optimum_benchmark/backends/pytorch.py

```diff
@@ -72,10 +73,88 @@ class PyTorchConfig(BackendConfig):
+
+    def __post_init__(self):
+        """
+        Here we perform checks and transformations on the config.
+        But we never modify the types of the config values.
+        """
+
+        CUDA_VISIBLE_DEVICES = os.environ.get("CUDA_VISIBLE_DEVICES", None)
+
+        if self.torch_compile:
+            self.torch_compile_kwargs = OmegaConf.merge(
+                self.torch_compile_kwargs
+                if self.torch_compile_kwargs is not None
+                else {},
+                DEFAULT_COMPILE_CONFIG,
+            )
+
+        if self.device_map is not None:
+            assert self.device_map in ["auto", "sequential"], (
+                "`device_map` must be one of ['auto', 'sequential']. "
+                "are supported in Optimum-Bnechmark. "
+                f"Got {type(self.device_map)} instead."
+            )
+            assert (
+                CUDA_VISIBLE_DEVICES is not None
+            ), "`device_map` can only be used when CUDA_VISIBLE_DEVICES is set."
+
+        if self.torch_dtype is not None:
+            assert self.torch_dtype in ["bfloat16", "float16", "float32", "auto"], (
+                "`torch_dtype` must be one of ['bfloat16', 'float16', 'float32', "
+                f"'auto']. Got {self.torch_dtype} instead."
+            )
+
+        if self.amp_dtype is not None:
+            assert self.amp_dtype in ["bfloat16", "float16", "float32"], (
+                "`amp_dtype` must be one of ['bfloat16', 'float16', 'float32']. "
+                f"Got {self.amp_dtype} instead."
+            )
+
+        if self.quantization_strategy is not None:
+            assert self.quantization_strategy in ["bnb", "gptq"], (
+                "`quantization_strategy` must be one of ['bnb', 'gptq']. "
+                f"Got {self.quantization_strategy} instead."
+            )
+            if self.quantization_strategy == "gptq":
+                bits = self.quantization_config.get("bits", None)
+                assert bits is not None, (
+                    "`quantization_config.bits` must be provided "
+                    "when using 'gptq' quantization strategy."
+                )
+        else:
+            self.quantization_config = None
+
+        if self.use_ddp:
+            self.ddp_config = OmegaConf.merge(
+                self.ddp_config if self.ddp_config is not None else {},
+                DEFAULT_DDP_CONFIG,
+            )
+
+            # TODO: support multi-node training.
+            assert self.ddp_config.max_nodes == 1, (
+                "Currently, PyTorch DDP training benchmark "
+                "only supports training on a single node."
+            )
+
+            assert (
+                CUDA_VISIBLE_DEVICES is not None
+            ), "Pytorch DDP training benchmark requires CUDA_VISIBLE_DEVICES to be set."
+        else:
+            self.ddp_config = None
+
 
 class PyTorchBackend(Backend):
+    name: str = "pytorch"
+    config: PyTorchConfig
```

Comment:
Why is config a class attribute and not an instance attribute? It is probably fine since we always have a single instance of `PyTorchBackend`, but it can be bug prone otherwise.

Reply:
it is an instance attribute without a default value. the only class attribute is the name.